### PR TITLE
Add Koala Script as new analytics integration

### PR DESCRIPTION
**Pull request by hanywang2**
## Summary

Quite meta. The docs to build the script was on our own [docs site](https://koala.mintlify.app/get-started/quick-start).

### CODE REVIEW

File: client/src/analytics/AbstractAnalyticsImplementation.ts [hunk 1]
```diff
@@ -50,6 +54,7 @@ export type ConfigInterface = AmplitudeConfigInterface &
   MixpanelConfigInterface &
   PirschConfigInterface &
   PostHogConfigInterface &
+  PlausibleInterface &
```
Comment (tobias-mintlify) [inline client/src/analytics/AbstractAnalyticsImplementation.ts:57]: I think you meant to add KoalaInterface

File: packages/validation/src/schemas/analytics.ts [hunk 2]
```diff
@@ -147,6 +147,20 @@ const plausibleConfigInterfaceSchema = z.object(
   }
 );
 
+const koalaConfigInterfaceSchema = z.object(
+  {
+    projectId: z
+      .string({
+        required_error: "Project ID is required for the snippet to run.",
+        invalid_type_error: "Koala Project ID must be a string.",
+      })
```
Comment (tobias-mintlify) [inline packages/validation/src/schemas/analytics.ts:156]: is there a minimum length? or do tokens always start with `id_` or something similar? Just looking to make the validation more specific. Docs:  https://github.com/colinhacks/zod#strings

**Review by tobias-mintlify — CHANGES_REQUESTED** [review]

File: client/src/analytics/AbstractAnalyticsImplementation.ts (hunk 1 above)
Comment (hanywang2) [reply client/src/analytics/AbstractAnalyticsImplementation.ts:57]: oop u right

File: client/src/analytics/KoalaScript.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import Script from 'next/script';
+
+import { KoalaInterface } from './AbstractAnalyticsImplementation';
+
+export default function KoalaScript({ koala }: { koala?: KoalaInterface }) {
+  if (!koala?.projectId || process.env.NODE_ENV !== 'production') {
+    return null;
+  }
+
+  // There is no npm package for Google Analytics 4 so set up happens by placing this script.
```
Comment (tobias-mintlify) [inline client/src/analytics/KoalaScript.tsx:10]: Copy-paste error, "Google Analytics 4"

**Review by tobias-mintlify — CHANGES_REQUESTED** [review]

**Review by tobias-mintlify — APPROVED** [review]

File: client/src/analytics/AbstractAnalyticsImplementation.ts [hunk 4]
```diff
@@ -50,7 +54,8 @@ export type ConfigInterface = AmplitudeConfigInterface &
   MixpanelConfigInterface &
   PirschConfigInterface &
   PostHogConfigInterface &
-  PlausibleInterface;
+  PlausibleInterface &
+  KoalaInterface;
```
Comment (benjamin-shen) [inline client/src/analytics/AbstractAnalyticsImplementation.ts:58]: 🔤

File: client/src/analytics/AnalyticsMediator.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ import {
   PirschConfigInterface,
   GoogleTagManagerConfigInterface,
   PlausibleInterface,
+  KoalaInterface,
```
Comment (benjamin-shen) [inline client/src/analytics/AnalyticsMediator.ts:14]: 🔤

File: client/src/analytics/AnalyticsMediator.ts [hunk 6]
```diff
@@ -33,6 +34,7 @@ export type AnalyticsMediatorConstructorInterface = {
   pirsch?: PirschConfigInterface;
   posthog?: PostHogConfigInterface;
   plausible?: PlausibleInterface;
+  koala?: KoalaInterface;
```
Comment (benjamin-shen) [inline client/src/analytics/AnalyticsMediator.ts:37]: 🔤

File: client/src/layouts/SupremePageLayout.tsx [hunk 7]
```diff
@@ -113,6 +114,7 @@ export default function SupremePageLayout({
             <GA4Script ga4={analyticsConfig.ga4} />
             <GTMScript gtm={analyticsConfig.gtm} />
             <PlausibleScript plausible={analyticsConfig.plausible} />
+            <KoalaScript koala={analyticsConfig.koala} />
```
Comment (benjamin-shen) [inline client/src/layouts/SupremePageLayout.tsx:117]: 🔤

File: packages/validation/src/schemas/analytics.ts [hunk 8]
```diff
@@ -147,6 +147,24 @@ const plausibleConfigInterfaceSchema = z.object(
   }
 );
 
+const koalaConfigInterfaceSchema = z.object(
```
Comment (benjamin-shen) [inline packages/validation/src/schemas/analytics.ts:150]: 🔤

File: packages/validation/src/schemas/analytics.ts [hunk 9]
```diff
@@ -159,7 +177,8 @@ export const analyticsSchema = z
     pirsch: pirschConfigInterfaceSchema.optional(),
     posthog: postHogConfigInterfaceSchema.optional(),
     plausible: plausibleConfigInterfaceSchema.optional(),
+    koala: koalaConfigInterfaceSchema.optional(),
```
Comment (benjamin-shen) [inline packages/validation/src/schemas/analytics.ts:180]: 🔤

File: packages/validation/src/schemas/analytics.ts [hunk 10]
```diff
@@ -159,7 +177,8 @@ export const analyticsSchema = z
...
   })
   .strict(
-    "Mintlify only supports analytics integrations from: amplitude, fathom, ga4, gtm, hotjar, logrocket, mixpanel, pirsch, posthog, and plausible."
+    "Mintlify only supports analytics integrations from: amplitude, fathom, ga4, gtm, hotjar, logrocket, mixpanel, pirsch, posthog, plausible, and koala."
```
Comment (benjamin-shen) [inline packages/validation/src/schemas/analytics.ts:183]: 🔤

File: client/src/analytics/AbstractAnalyticsImplementation.ts [hunk 11]
```diff
@@ -40,6 +40,10 @@ export type PlausibleInterface = {
   domain?: string;
 };
 
+export type KoalaInterface = {
```
Comment (benjamin-shen) [inline client/src/analytics/AbstractAnalyticsImplementation.ts:43]: 🔤

**Review by benjamin-shen — COMMENTED** [review]
@hanywang2 nit: I think we should keep these alphabetized as we add them (mostly for organization, but especially for the validation error message). I marked everywhere out-of-order with an 🔤

**Pull request merged**